big_scape/network/network.py

```diff
@@ -324,7 +385,65 @@ def get_random_edge(
 
     if temp_record_table is not None:
         random_edge_query = random_edge_query.where(
+            and_(
+                distance_table.c.record_a_id.in_(select(temp_record_table.c.record_id)),
+                distance_table.c.record_b_id.in_(select(temp_record_table.c.record_id)),
+            )
+        )
+
+    edge = DB.execute(random_edge_query).fetchone()
+
+    return edge
+
+
+def get_random_edge_seeded(
```

Comment:
it feels like this can be merged with get_random_edge in some kind of way. but I'll do that after this branch is merged